### PR TITLE
drivers: ethernet: enc28j60: Fix spi transaction on read regs

### PR DESCRIPTION
**Pull request by dean-at-evos**
Some SPI controllers (eg: esp32) could break the transaction when reading the dummy byte registers into two. This caused the read to fail.

### CODE REVIEW

**Review by nordicjm — COMMENTED** [review]
@tbursztyka @teburd isn't this a bug in the esp32 driver not the ethernet driver? I can see similar code to how this works before this change in lots of drivers which seems to work fine for all other chips.

**Comment by dean-at-evos** [comment]
> @tbursztyka @teburd isn't this a bug in the esp32 driver not the ethernet driver? I can see similar code to how this works before this change in lots of drivers which seems to work fine for all other chips.

@nordicjm has a point, this does expose a bug in the ESP32 master driver. I have a couple of PR on this as well.
I probably should also have said it is a small bw efficiency gain. Not having the buffers as the same length/count results in a context switch mid transaction (I am away from the hware and logic analyser to post examples), the ESP driver probably mistakenly also cycles the CS, but any implementation will take a small hit on bus bandwidth without this PR. The trade off is bytes of stack space vs code space for the tx buf.

**Comment by nordicjm** [comment]
I'll give this a try when I have time on a nordic board and time it so we can compare, we should not be merging things that are specifically tailored to a single MCU which worsens them on other MCUs.

CC @sylvioalves for the esp32 SPI issue.

**Comment by dean-at-evos** [comment]
> I'll give this a try when I have time on a nordic board and time it so we can compare, we should not be merging things that are specifically tailored to a single MCU which worsens them on other MCUs.
> 
> CC @sylvioalves for the esp32 SPI issue.
TLDR; This PR improves bandwidth anyway, doesn't FIX the driver except for some SOC.

OK, so a bit of poking around and I have found this issue #54199 that may or not be related, and there are probably others sleeping I reckon .... 
This chip has a couple of weird timing and register access requirements. The one that has caught me up and exposed intermittent bugs is that for a subset of registers the /CS hold time is quite long...extract from the data sheet;
image
In the case where your board/device tree setup uses the GPIO as chip select it probably doesn't matter as there is enough delay in setting up context that masks the issue. Where drivers use a hardware driven CS, things got funky for me(on ESP32 but I don't think it matters which SOC). I would see all sorts of inconsistent behavior, and probably annoyed the network admins with crappy malformed junk on the wire....
This led to me putting in the PR on the ESP32 controller driver that allows setting the CS timing, #57570,  where there are a couple of logic analyser shots that have timing. In that case I didn't capture the cases of where the driver fails but until I hooked up the LA and watched the bus traffic all sorts of working/not working/crashing was happening. Extending the /CS hold time fixed it all, I could ping the SOC over hours reliably.

**Review by tbursztyka — CHANGES_REQUESTED** [review]
Which controllers?

Controllers are supposed to behave the same when accessed through the generic Zephyr SPI API. If some are not behaving properly, they will need to be fixed.
